pandas/io/formats/format.py

```diff
@@ -696,9 +696,11 @@ def _truncate_vertically(self) -> None:
         assert self.max_rows_fitted is not None
         row_num = self.max_rows_fitted // 2
         if row_num >= 1:
-            head = self.tr_frame.iloc[:row_num, :]
-            tail = self.tr_frame.iloc[-row_num:, :]
-            self.tr_frame = concat((head, tail))
+            _len = len(self.tr_frame)
+            rows = np.arange(_len)
+            self.tr_frame = self.tr_frame.iloc[
+                (rows < row_num) | (rows >= _len - row_num)
```

Comment:
this is going to allocate three masks.  can we avoid that?

Reply:
we could use 
```python
head = [ i for i in range(row_num)]
tail = [_len - row_num + i for i in range(row_num)]
self.tr_frame.iloc[head + tail]
```
it would probably be faster for reasonable `row_num`

Reply:
i implemented this in the newest commit